src/main/kotlin/com/pubnub/api/v2/entities/Channel.kt

```diff
@@ -0,0 +1,39 @@
+package com.pubnub.api.v2.entities
+
+import com.pubnub.api.v2.subscriptions.Subscription
+import com.pubnub.api.v2.subscriptions.SubscriptionOptions
+
+/**
+ * A representation of a PubNub channel identified by its [name].
+ *
+ * You can get a [Subscription] to this channel through [subscription].
+ *
+ * Use the [com.pubnub.api.PubNub.channel] factory method to create instances of this interface.
+ */
+interface Channel : Subscribable {
+    /**
+     * The name of this channel. Supports wildcards by ending it with ".*"
+     *
+     * See more in the [documentation](https://www.pubnub.com/docs/general/channels/overview)
+     */
+    val name: String
+
+    /**
+     * Returns a [Subscription] that can be used to subscribe to this channel.
+     *
+     * Channel subscriptions support passing [com.pubnub.api.v2.subscriptions.ChannelOptions.receivePresenceEvents] in
+     * [options] to enable receiving presence events.
+     *
+     * The default [SubscriptionOptions] such as [SubscriptionOptions.filter] can also be used to filter events
+     * delivered to the subscription.
+     *
+     * For example, to create a subscription that only listens to presence events:
+     * ```
+     * channel.subscription(ChannelOptions.receivePresenceEvents() + SubscriptionOptions.filter( { it is PNPresenceEventResult } ))
```

Comment:
I am thinking how user may know what class filter by?
In this example there is PNPresenceEventResult but how to get know which other class can be used here for filtering.
Maybe we could introduce enum with available classes?

Reply:
added a documentation note on Filter and will explore making PNEvent sealed

Reply:
unfortunately PNEvent can't be sealed because our event classes are in different packages and it's not allowed :( 
shame, but I don't think we have a choice. I've updated the doc and if the user wants to filter for events only of some class then they probably already know what class they want anyway, so I don't think it's a big deal. And they can always check `EventEmitter`